R/readWriteMgfData.R

```diff
@@ -20,13 +20,28 @@ setMethod("writeMgfData",
                              verbose = verbose)
           })
 
+#' @param addFields `data.frame` or `matrix` with optional additional
+#'     fields to be added to each spectrum (each column one field).
+#'
+#' @noRd
 writeMgfDataFile <- function(splist, con, COM = NULL, TITLE = NULL,
-                             verbose = isMSnbaseVerbose()) {
+                             verbose = isMSnbaseVerbose(), addFields = NULL) {
   if (class(con) == "character" && file.exists(con)) {
     message("Overwriting ", con, "!")
     unlink(con)
   }
 
+  if (length(addFields)) {
+      if (is.null(dim(addFields)))
+          stop("'addFields' has to be a matrix or data.frame.")
+      else addFields <- as.matrix(addFields, ncol = ncol(addFields))
+      if (is.null(colnames(addFields)))
+          stop("Column names required on 'addFields'.")
+      if (nrow(addFields) != length(splist))
+          stop("nrow of 'addFields' has to match length of 'splist'")
+      haveAddFields <- TRUE
```

Comment:
Do you need this additional variable `haveAddFields`? Would not `is.null(addFields)` be enough in line 63 (`if (haveAddFields)`). But even this would not be needed at all (see below).

Reply:
No, we don't. Usually I add a variable instead of calling `is.null` 2 or more times on the same object in a function. Doesn't apply here, but if the function would be called in a loop this can have a performance effect.
But good point with the `NULL` subsetting! Actually never tried subsetting `NULL` 😄